HACKING.md

```diff
@@ -0,0 +1,95 @@
+# Hacking
+
+## Quickstart
+
+Use [ghcup][ghcup] to install the `cabal` cli tool and the ghc version we're using:
+
+```sh
+$ ghcup install ghc 8.10
+$ ghcup set ghc 8.10
+$ ghcup install cabal 
+$ ghcup set cabal
+$ cabal update
+$ cabal build
+```
+
+### Building
+
+In the base directory, run `cabal build`
+
+### Running tests
+
+In the base directory, run `cabal test`
+
+## Tools
+
+| name | description |
+| ---- | ----------- |
+| [ghcup][ghcup] | Used to manage installed versions of ghc and cabal-install |
+| ghc | The haskell compiler (installed via ghcup) |
+| cabal-install | The package manager we use (installed via ghcup) |
+| [haskell-language-server][hls] | LSP server for haskell projects |
+| [hlint][hlint] | A linting + hints tool for haskell code. It provides really useful suggestions |
```

Comment:
It might be helpful to mention that hlint, ormolu, and fourmolu are bundled with haskell-language-server

```suggestion
| [hlint][hlint] | A linting + hints tool for haskell code. It provides really useful suggestions (bundled with haskell-language-server) |
```